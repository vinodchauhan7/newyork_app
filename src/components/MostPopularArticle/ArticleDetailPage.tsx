import React from 'react';
import { useLocation } from 'react-router-dom';
import { getMediaPicture } from '../../lib/utils';

const ArticleDetailPage = () => {
    let { state } = useLocation();
    return (
        <div className='flex flex-col md:flex-row justify-center gap-2 mt-8' data-testid="articleCardDetail">
            <div className='w-full md:w-1/2 h-1/2 bg-gray-200'>
            <img src={getMediaPicture(state.data?.media)?.url} alt={state.data?.title} className='w-full h-full object' />
            </div>
            <div className='w-full md:w-1/2 h-1/2'>
                <h1 className='font-bold text-2xl'>{state.data.title}</h1>
                <div className='flex-grow border-t-2 border-black'></div>
                <p className='font-light text-xl'>{state.data.abstract}</p>
            </div> 

        </div>
    )
}

export default ArticleDetailPage;