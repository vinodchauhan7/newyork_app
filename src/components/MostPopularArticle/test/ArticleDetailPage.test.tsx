
import ArticleDetailPage from "../ArticleDetailPage"
import { render} from "@testing-library/react";
import '@testing-library/jest-dom';
import {MemoryRouter } from "react-router-dom";


test("Render articles Component", () => {
  const {container} = render(
    <MemoryRouter initialEntries={[{ pathname: '/article:id', state: {data: {}} }]}>
       <ArticleDetailPage/>
      </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
})