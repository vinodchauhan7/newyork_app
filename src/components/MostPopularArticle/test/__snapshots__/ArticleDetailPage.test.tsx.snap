// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render articles Component 1`] = `
<div>
  <div
    class="flex justify-center gap-2"
    data-testid="articleCardDetail"
  >
    <div
      class="w-1/2 h-1/2 bg-gray-200"
    >
      <img
        class="w-full h-full object"
      />
    </div>
    <div
      class="w-1/2 h-1/2"
    >
      <h1
        class="font-bold text-2xl"
      />
      <div
        class="flex-grow border-t-2 border-black"
      />
      <p
        class="font-light text-xl"
      />
    </div>
  </div>
</div>
`;
